services/app-api/src/postgres/user/insertManyUsers.ts

```diff
@@ -0,0 +1,42 @@
+import { PrismaClient } from '@prisma/client'
+import { StoreError, convertPrismaErrorToStoreError } from '../storeError'
+import { UserType, toDomainUser } from '../../domain-models'
+import { InsertUserArgsType } from './insertUser'
+
+export async function insertManyUsers(
+    client: PrismaClient,
+    users: InsertUserArgsType[]
+): Promise<UserType[] | StoreError> {
+    try {
+        console.info(`Trying to insert ${users.length} users into postgres....`)
+
+        const createResult = await client.user.createMany({
+            data: users.map((user) => ({
+                id: user.userID,
+                givenName: user.givenName,
+                familyName: user.familyName,
+                email: user.email,
+                role: user.role,
+                stateCode: user.stateCode ?? null,
+                divisionAssignment: user.divisionAssignment ?? null,
+            })),
+            skipDuplicates: true,
+        })
+
+        console.info(
+            `Insert users completed, ${createResult.count} of ${users.length} users inserted.`
+        )
+
+        const userIDs = users.reduce((a: string[], c) => a.concat(c.userID), [])
```

Comment:
I think I'd go with
```javascript
const userIDs = users.map((user) => user.userID);
```
here.

Reply:
I must have been delirious when I wrote that. Thanks for the catch.